[1.15.x] Added entity nameplate rendering event hook

LeScooter opened the pull request:
Adds a new event hook RenderNameplateEvent.

Commit:
- Added entity nameplate rendering event hook

Commit:
- Added entity nameplate rendering event hook

diesieben07 commented on src/main/java/net/minecraftforge/client/event/RenderNameplateEvent.java:
```diff
@@ -0,0 +1,67 @@
+     */
+    public String getOriginalContent()
+    {
+        return this.nameplateContent;
```
Should be `originalContent`.

LeScooter replied:
Now fixed. Thanks

diesieben07 commented on src/main/java/net/minecraftforge/client/event/RenderNameplateEvent.java:
```diff
@@ -0,0 +1,67 @@
+ * {@link #originalContent} contains the original content being rendered on the name plate/tag. This cannot be changed by mods.<br>
+ * <br>
+ * This event {@link HasResult}. <br>
+ * This event has a result. ALLOW will force-render name plate/tag, & DENY will prevent name plate/tag from rendering<br>
```
And what about `DEFAULT`?

LeScooter replied:
Now added DEFAULT use to class doc

diesieben07 commented on patches/minecraft/net/minecraft/client/renderer/entity/EntityRenderer.java.patch:
```diff
@@ -0,0 +1,23 @@
++      net.minecraftforge.client.event.RenderNameplateEvent renderNameplateEvent = new net.minecraftforge.client.event.RenderNameplateEvent(p_225623_1_,p_225623_1_.func_145748_c_().func_150254_d());
++      net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(renderNameplateEvent);
++
++      if (!renderNameplateEvent.getResult().equals(Event.Result.DENY) && (renderNameplateEvent.getResult().equals(Event.Result.ALLOW) || this.func_177070_b(p_225623_1_))) {
```
Just use `==` for enums.

diesieben07 replied:
Also, this doesn't even compile, because `Event` is not a thing here... How on earth did you generate this patch?

LeScooter replied:
Replaced equals with == operator now

Commit:
- Updated RenderNameplateEvent.java doc
- Fixed getOriginalContent() not returning original content
- Replaced .equals() with == operator

diesieben07 commented on patches/minecraft/net/minecraft/client/renderer/entity/EntityRenderer.java.patch:
```diff
@@ -0,0 +1,23 @@
++      if (renderNameplateEvent.getResult() != Event.Result.DENY && (renderNameplateEvent.getResult() == Event.Result.ALLOW || this.func_177070_b(p_225623_1_))) {
++         this.func_225629_a_(p_225623_1_, renderNameplateEvent.getContent(), p_225623_4_, p_225623_5_, p_225623_6_);
+       }
++
```
No whitespace changes in patches.
Still produces invalid code, because `Event` is not a thing in this class.

diesieben07 commented on src/main/java/net/minecraftforge/client/event/RenderNameplateEvent.java:
```diff
@@ -0,0 +1,68 @@
+    {
+        this.entity = entity;
+        this.setContent(content);
+        this.originalContent = Strings.nullToEmpty(content);
```
Why are we silently converting null to empty?

diesieben07 commented on src/main/java/net/minecraftforge/client/event/RenderNameplateEvent.java:
```diff
@@ -0,0 +1,68 @@
+     */
+    public void setContent(String contents)
+    {
+        this.nameplateContent = Strings.nullToEmpty(contents);
```
Again, why nullToEmpty?

diesieben07 commented on src/test/java/net/minecraftforge/debug/client/rendering/NameplateRenderingEventTest.java:
```diff
@@ -0,0 +1,35 @@
+
+    }
+
+}
```
This file doesn't follow the code formatting guidelines.

Commit:
- Removed nullToEmpty
- Fixed patch
- Removed import from patch

Commit:
- Removed nullToEmpty
- Fixed patch
- removed

Commit:
- Removed unnecessary nullToEmpty

Commit:
- Fixed formatting to comply with guidelines

Commit:
- Fixed formatting issue in nameplate rendering event test class

ichttt commented on src/main/java/net/minecraftforge/client/event/RenderNameplateEvent.java:
```diff
@@ -0,0 +1,67 @@
+ * This event is fired on the {@link MinecraftForge#EVENT_BUS}.
+ **/
+@Event.HasResult
+public class RenderNameplateEvent extends Event
```
Should extend EntityEvent instead of the raw Event class

ichttt commented on src/main/java/net/minecraftforge/client/event/RenderNameplateEvent.java:
```diff
@@ -0,0 +1,67 @@
+    /**
+     * The entity in which the name plate/tag is being rendered to
+     */
+    public Entity getEntity()
```
Not needed when extending EntityEvent

Commit:
- RenderNameplateEvent now extends EntityEvent
- Removed unnecessary entity getter

ichttt commented on src/main/java/net/minecraftforge/client/event/RenderNameplateEvent.java:
```diff
@@ -0,0 +1,58 @@
+    private String nameplateContent;
+    private final String originalContent;
+
+    public RenderNameplateEvent(Entity entity, String content)
```
You should also pass in the MatrixStack and the IRenderTypeBuffer to the event, in case anyone wants to render something special on the nameplate

LeScooter replied:
Added

Commit:
- Added MatrixStack and IRenderTypeBuffer to event

Commit:
- Removed unnecessary white space
